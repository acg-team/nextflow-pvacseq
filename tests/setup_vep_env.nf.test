nextflow_workflow {

    name "Test Subworkflow SETUP_VEP_ENVIRONMENT"
    script "subworkflows/local/setup_vep_env/main.nf"
    workflow "SETUP_VEP_ENVIRONMENT"

    test("Both cache and plugins directories exist") {

        when {
            params {
                vep_cache = "$projectDir/tests/data/vep_cache"
                vep_plugins = "$projectDir/tests/data/VEP_plugins"
                vep_cache_version = "102"
                vep_genome = "GRCh38"
                vep_species = "homo_sapiens"
            }
            workflow {
                """
                input[0] = "$params.vep_cache"
                input[1] = "$params.vep_cache_version"
                input[2] = "$params.vep_genome"
                input[3] = "$params.vep_species"
                input[4] = "$params.vep_plugins"
                """
            }
        }

        then {
            assert workflow.success
            assert (workflow.out.vep_cache).get(0) == "$params.vep_cache"
            assert (workflow.out.vep_plugins).get(0) == "$params.vep_plugins"
        }

    }

    test("Download cache and plugins") {

        when {
            params {
                vep_cache = []
                vep_plugins = []
                vep_cache_version = "113"
                vep_genome = "WBcel235"
                vep_species = "caenorhabditis_elegans"
            }
            workflow {
                """
                input[0] = []
                input[1] = "$params.vep_cache_version"
                input[2] = "$params.vep_genome"
                input[3] = "$params.vep_species"
                input[4] = []
                """
            }
        }

        then {
            assert workflow.success

            def stable_path_cache = getAllFilesFromDir(workflow.out.vep_cache.get(0).get(0), ignoreFile: 'tests/.nftignore')
            def stable_path_plugins = getAllFilesFromDir(workflow.out.vep_plugins.get(0).get(0), ignoreFile: 'tests/.nftignore')
            assert snapshot(
                    workflow.out.vep_cache_version,
                    workflow.out.vep_genome,
                    workflow.out.vep_species,
                    // All files with stable contents
                    stable_path_cache,
                    stable_path_plugins
            ).match()
        }

    }
}
