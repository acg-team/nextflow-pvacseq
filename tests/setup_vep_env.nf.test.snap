{
    "Download cache and plugins": {
        "content": [
            [
                "113"
            ],
            [
                "WBcel235"
            ],
            [
                "caenorhabditis_elegans"
            ],
            [
                "1-1000000.gz:md5,cadcba92b0999210dd8d832505d2e4c4",
                "10000001-11000000.gz:md5,998a75dd927d10d45f8eebeef5fc7a75",
                "1000001-2000000.gz:md5,a5cb3adb1ec9f40eed6a355d1492ba9b",
                "11000001-12000000.gz:md5,46e6917f51093e28cce061774b9ed158",
                "12000001-13000000.gz:md5,0adffacf8482d6c224df27104f65c9d6",
                "13000001-14000000.gz:md5,aee759d812fc900a980ab0c4c5bd0273",
                "14000001-15000000.gz:md5,f65537a3f76c40e63b6deb0b6cdb09dc",
                "15000001-16000000.gz:md5,379f092ad1afa888da1fc13e80535def",
                "2000001-3000000.gz:md5,86839741524579fd089498d6bee44dff",
                "3000001-4000000.gz:md5,509b28af3920427e951f00b6973b5df4",
                "4000001-5000000.gz:md5,f606e69cf59b0bdf2b61653608d955a6",
                "5000001-6000000.gz:md5,a14ce1e21856e4a77ed63c67cbdfb26a",
                "6000001-7000000.gz:md5,e1a895d6e8b352182b53ed1d0ce6e24e",
                "7000001-8000000.gz:md5,ddf91b60f636d26b68b6bab3520b6b32",
                "8000001-9000000.gz:md5,57482b996f89e92bbd0196efa4915cd3",
                "9000001-10000000.gz:md5,43b5d89f84236b49b384d7f37f928129",
                "1-1000000.gz:md5,d18811781848f70baef0b0348190d7ce",
                "10000001-11000000.gz:md5,19011165abc56233ea0c5b0e6938d9c9",
                "1000001-2000000.gz:md5,5e720fa191f3c9ac799b6a071bcc4332",
                "11000001-12000000.gz:md5,b19c46fb00ca13a2a31128bd1829ddf5",
                "12000001-13000000.gz:md5,54354b0870ca96641c51ed63382da007",
                "13000001-14000000.gz:md5,6954fdc223f58eb406e602752ab7d139",
                "14000001-15000000.gz:md5,929275a1cfea883999dddc20931a2e72",
                "15000001-16000000.gz:md5,5f5b783a589a1fd80cc565e6f339c540",
                "2000001-3000000.gz:md5,54e476e0e9f4a5d973ee710fd824abc7",
                "3000001-4000000.gz:md5,d78d4a63165429fdb3a61b7cdbd3c43a",
                "4000001-5000000.gz:md5,983f8efcebb7f62d7e7b1b3c0573d43e",
                "5000001-6000000.gz:md5,e2cd03ed5b67b8ee123e4c4958508fe4",
                "6000001-7000000.gz:md5,d04bc9335ba39ace20bce936e3a5cdeb",
                "7000001-8000000.gz:md5,9354b26a9ba94aa5bc30f537c22382fb",
                "8000001-9000000.gz:md5,b227c6ef81ab72d211d25dc4f44813b9",
                "9000001-10000000.gz:md5,a6d7f29edd7c22139403a11cac989b7a",
                "1-1000000.gz:md5,2117acb322a117a9c5db85c072575331",
                "10000001-11000000.gz:md5,646c9582b56eb12ddbb1dd35b25c3670",
                "1000001-2000000.gz:md5,ee433e4e5e37b2d008c43e1af4be0f8d",
                "11000001-12000000.gz:md5,962fd6e52046484b3b123f9380ed64e9",
                "12000001-13000000.gz:md5,1abf2d695c829eb2c88e0d3dbc739a1c",
                "13000001-14000000.gz:md5,a6e03bf867f5cc694174a230f1b13a6b",
                "2000001-3000000.gz:md5,a5b250aa9e3ee8cecc23bea0e2fa19a1",
                "3000001-4000000.gz:md5,1390a6d2a28a4861b282d36d0fb85660",
                "4000001-5000000.gz:md5,4bc7106bb2661aea28613c31935a5c8f",
                "5000001-6000000.gz:md5,7317d6fbb3c77d7cdd31e781afab8f7d",
                "6000001-7000000.gz:md5,1a3b6fa586e570c16b4833e34b28751e",
                "7000001-8000000.gz:md5,b7bcb06393682f621403afdf19bf87b4",
                "8000001-9000000.gz:md5,0011675a8567d394da54a52480b35786",
                "9000001-10000000.gz:md5,e4fa88e4ec57ed0c71fd21090d8aa17a",
                "1-1000000.gz:md5,a47af22d33275652036ddf7161699c7c",
                "10000001-11000000.gz:md5,7fc129e7edbaa5be87306de417c2ef28",
                "1000001-2000000.gz:md5,cbc12c339741df5ad06bf9a946be6c93",
                "11000001-12000000.gz:md5,d1cc5e20e3d3402debdc102087a5407f",
                "12000001-13000000.gz:md5,42c69c8e86d28151e9a8b1787dbee125",
                "13000001-14000000.gz:md5,c7459d1789a833e8a898ebdbc607e7d8",
                "14000001-15000000.gz:md5,5806b20108f56d9eeabcdd4f8450dca3",
                "15000001-16000000.gz:md5,78e859f70026a05be43d48b9b272f287",
                "16000001-17000000.gz:md5,539db7fc976bee4b6031f8dcb6a4641d",
                "17000001-18000000.gz:md5,f3ea55e7552dc36734d6e8ba67d1e4c2",
                "2000001-3000000.gz:md5,539013ecfdcd06eb653445f857265322",
                "3000001-4000000.gz:md5,beb9701b402bd5ddc46a4da6e531f783",
                "4000001-5000000.gz:md5,3f46efb2635850cc6c3d8ae51727a400",
                "5000001-6000000.gz:md5,e11549bca12c5e2a7a208a997fda1c68",
                "6000001-7000000.gz:md5,c0f3546c6859dc1a5fe9ff7f015ecd7e",
                "7000001-8000000.gz:md5,344b72822f647819f4ee6b5afa9d7701",
                "8000001-9000000.gz:md5,1c06d285ff5c53f89f073212343902b7",
                "9000001-10000000.gz:md5,79140e754039c6d6fc6eeecddcf2aa8e",
                "1-1000000.gz:md5,40ef48190d3269cd4112450bc717b1ef",
                "1-1000000.gz:md5,1a8739457c429931923ed77596a9ee54",
                "10000001-11000000.gz:md5,316fa1d06fc1878b6a5995f4aee3e49d",
                "1000001-2000000.gz:md5,3926c03a091850c909bd0ccfc7133c0b",
                "11000001-12000000.gz:md5,29ca11d2f05051cc439a0d24a9db134c",
                "12000001-13000000.gz:md5,a46f648554e91999652019516c933754",
                "13000001-14000000.gz:md5,167b126d1c690a0e7e25fc5ccd09fb7c",
                "14000001-15000000.gz:md5,645554c896133c476c3083302371bcf8",
                "15000001-16000000.gz:md5,60fc48d9a7aff6286fc6630c46bcfebc",
                "16000001-17000000.gz:md5,07e1750d1c95a61e96774d2cf3da4d89",
                "17000001-18000000.gz:md5,59d084309f6a975ec1066a828b5845ba",
                "18000001-19000000.gz:md5,868c12d305dbd4d04399ec7848804328",
                "19000001-20000000.gz:md5,6de03a00061f6a88dcbbb8ed5fc0b8dc",
                "20000001-21000000.gz:md5,732b956f13da9ef01f9de3355d12e28b",
                "2000001-3000000.gz:md5,7c7528266c523cad419ea25e75d9566e",
                "3000001-4000000.gz:md5,bb2283c0cfb0e4601fc535a4d51e6f2d",
                "4000001-5000000.gz:md5,64c7f28f554414a88c886b0bcadb3c39",
                "5000001-6000000.gz:md5,58e7106fe577a8b5e5c698445b4f0c33",
                "6000001-7000000.gz:md5,2e309d12cf1c1c6276585f457ceeacc2",
                "7000001-8000000.gz:md5,08cb0600f7806608f0103187a6c9c64e",
                "8000001-9000000.gz:md5,869333c2615f714860d17d794640d4ad",
                "9000001-10000000.gz:md5,b85cc861c6a3b30cf6f06c8af136b383",
                "1-1000000.gz:md5,c8d97b084c159c3cb5be1fff4637dfce",
                "10000001-11000000.gz:md5,f441f2af06fd4973749dfbfbef40fe1b",
                "1000001-2000000.gz:md5,c42a1526a836cfacefb67e9217f648aa",
                "11000001-12000000.gz:md5,264421c249c696b45c92e2611285fee7",
                "12000001-13000000.gz:md5,e673d1fdbe7dc0d09bea3d11a5797d6a",
                "13000001-14000000.gz:md5,88f4f84e63b362f1b4f800c48b37e82c",
                "14000001-15000000.gz:md5,26282f2b305ed82fb9f8875e97361105",
                "15000001-16000000.gz:md5,30b9132c2610d42919ba231d1adbef2a",
                "16000001-17000000.gz:md5,3d0e975ccd1ae4e92bf1d9d915ed293f",
                "17000001-18000000.gz:md5,7db5b3819da3df1e47fe757dc9c6f2ba",
                "2000001-3000000.gz:md5,55f6130a8d5872bdc9f8eed231ad0f65",
                "3000001-4000000.gz:md5,402b826dbf6993c207ad15483a44182b",
                "4000001-5000000.gz:md5,43cf926d43db25af5724fb5077edfee1",
                "5000001-6000000.gz:md5,f40276dbea3f6f9a75f9301d1253eb09",
                "6000001-7000000.gz:md5,df0d2d38060d4e7c606072ae814b1f38",
                "7000001-8000000.gz:md5,c4117cc51255c0a91c51ff43403f00f7",
                "8000001-9000000.gz:md5,59a4ebadca27041634c58652c544c8dd",
                "9000001-10000000.gz:md5,c54510616273a4d1bfa9d525dbbbca40",
                "chr_synonyms.txt:md5,d390f0bcc6fec9786bc66b75f2d4390b",
                "info.txt:md5,249c88c7a71464e048cca0c4b2a21198"
            ],
            [
                ".hgignore:md5,593166ea263b7bb72ad26cb7cbe891c7",
                "AVADA.pm:md5,66b9fe2504ec1da195e8ecce0787b511",
                "AlphaMissense.pm:md5,de353cb0a717f7d52ae53d2b885278f3",
                "AncestralAllele.pm:md5,8c2340a005b4fc1c420b17a382330ad2",
                "BayesDel.pm:md5,9bc361aca23c8274d26769c45f2d9f2e",
                "Blosum62.pm:md5,e278604fd4c958fd008457ba3c197fd7",
                "CADD.pm:md5,b6365e388ea8beda043bccb7a2400887",
                "CAPICE.pm:md5,765574a8017dcdf95850425597fdef5c",
                "CCDSFilter.pm:md5,d261ba60077ef0d00f034331a1c3432d",
                "CONTRIBUTING.md:md5,e5a5b8a749f384de4883c48a2e6cc541",
                "CSN.pm:md5,ea6f6b62c13196a93b638461ee1f29b2",
                "Carol.pm:md5,0b5f43ae26c202f0f38a01878b5af389",
                "ClinPred.pm:md5,e743a809dbf053858ddbe699751adbbb",
                "Condel.pm:md5,093545533454fa72f75a09761032acb1",
                "Conservation.pm:md5,3d8afe604b786fb93162dc1ec8c01c8b",
                "DAS.pm:md5,bcb2376136493f194befe8f18b74976a",
                "DeNovo.pm:md5,ffedf0ea897229dbb947851af9c1b01f",
                "DosageSensitivity.pm:md5,832924ab64820331ec3ac8810ef135d8",
                "Downstream.pm:md5,5a381754483401c653016387b4f7817f",
                "Draw.pm:md5,6d818230baa8e6494543e38d64aa94fa",
                "EVE.pm:md5,c0a6f76a2d0ef2cf3b30da0123e39a51",
                "Enformer.pm:md5,9236c5b51f53c993493660b918f475c6",
                "FATHMM.pm:md5,5303c956add8c590ef4ea2b882fba605",
                "FATHMM_MKL.pm:md5,d2b03d95b27b8e31312a452de11a3b53",
                "FlagLRG.pm:md5,5ed9c6f80f09e7862e3425f4da6cad25",
                "Frameshift.pm:md5,7a8bd8e48f2b39a83232bac5400cf10f",
                "FunMotifs.pm:md5,c9a9848ac12b6c10068a7da65dffdf21",
                "G2P.pm:md5,f00ce3a28bd80cf0af0870ff8d802302",
                "GO.pm:md5,f2999675307d1f6798d2e96345b25e49",
                "GWAS.pm:md5,53d5ffbe1a5f70701ac0f70d1234fce9",
                "GXA.pm:md5,b52ef0422ea70519b56915c31512b2fe",
                "GeneBe.pm:md5,2f021880085e0e7eee6d34eb6386c618",
                "GeneSplicer.pm:md5,4ba802238ee9c1de47ba5c3e6fb6f5fd",
                "Geno2MP.pm:md5,beb17f0d125ffa8cf69c6f422a05cccb",
                "HGVSIntronOffset.pm:md5,b6b0be5223331024b0d08aa2d1689f46",
                "HGVSReferenceBase.pm:md5,7191d4e07685ae0135a632b5ddd74f02",
                "IntAct.pm:md5,0dfb619f02153f25c8f6179688418f6b",
                "LD.pm:md5,39da4f4f88845923cc61bf022a9d518f",
                "LICENSE:md5,4134cf9f4fab3a76cb923626eaff2238",
                "LOEUF.pm:md5,761977c2dea1670b7821461c4b10fd18",
                "LOVD.pm:md5,008ad91705aef9bef5ec175bf7885896",
                "LoFtool.pm:md5,a91b8c3cc77e2982864943388f1783c5",
                "LoFtool_scores.txt:md5,d9abd640ef22c790583b4b15d144ea6f",
                "LocalID.pm:md5,8c1e5fde0cec84f29d48337f6068a545",
                "MPC.pm:md5,683b1cfe518a9ee04b99be6dff7cbc99",
                "MTR.pm:md5,774a5b411eb7d480c17da19dd4875d01",
                "Mastermind.pm:md5,715e90e6d7a9bbe419789b2860fcdd50",
                "MaveDB.pm:md5,eff8d099bf12560cf2c9da2b235108a8",
                "MaxEntScan.pm:md5,ef163ae2eac23b0d511b13c752a04818",
                "NMD.pm:md5,2a5f79cafa518b5c21b85d41c22d7704",
                "NearestExonJB.pm:md5,27b0c790c4a03f6a53ed6baa51d6da21",
                "NearestGene.pm:md5,8d7d3d645283ccbc61b8d1b1c3b0e27b",
                "NonSynonymousFilter.pm:md5,02b29ea368036ef787035f68391f7912",
                "OpenTargets.pm:md5,106017e70dfc99d3b38b4a7d8af1ff1c",
                "PON_P2.pm:md5,12e139aa96a8c8a7cf13a5c020dde7a2",
                "Paralogues.pm:md5,ce4e79b6bedf01b33ae192bc5df2163b",
                "PhenotypeOrthologous.pm:md5,976697edcc2fda80c5ba91aca9ae0b0b",
                "Phenotypes.pm:md5,db112e0b1e0056dfc069fc2d9efd2b6e",
                "PolyPhen_SIFT.pm:md5,b1d6cdd27f4e570f15dd522ca5538742",
                "PostGAP.pm:md5,033898a81b30112bd083d780f3ab4ce1",
                "PrimateAI.pm:md5,0e5ed227348f7232e6caaad2da034e42",
                "ProteinSeqs.pm:md5,743ad4f8436d1a0aeb93099f2fab6238",
                "README:md5,1aee080c0c37543412c677ef7a5517dd",
                "README.md:md5,52fb50a6460faa228d8a3cde86c2318d",
                "REVEL.pm:md5,0b40d0d4c9e5eac1f3e50f1a2d5797fe",
                "RankFilter.pm:md5,f02e5ffad7c76d71e3c40a9108a91f79",
                "RefSeqHGVS.pm:md5,62ca83fb791b5fc88b6997cafd3c4c79",
                "ReferenceQuality.pm:md5,08f432fbf2f9a2b5fccc0cac9bf99b67",
                "RiboseqORFs.pm:md5,56f93c44fe988a98030d0c87f224da06",
                "SameCodon.pm:md5,51bd105ce92e6a353bd7141d1a332f11",
                "SingleLetterAA.pm:md5,e654effa3365cbf4c9a5206f2a4a4ebf",
                "SpliceAI.pm:md5,d1bcfd309d8da2a43ec72e2bd27338aa",
                "SpliceRegion.pm:md5,2084f411219ec1d7d9a0d11f6b32a0f7",
                "SpliceVault.pm:md5,0e36f60f94084477fb645e48ba0001a6",
                "StructuralVariantOverlap.pm:md5,729de6f54f2937a228934b3a18f2e4d6",
                "SubsetVCF.pm:md5,b1cdba93aa1408607e7d6d9fa28dd6c3",
                "TSSDistance.pm:md5,145d9514fb179f17d808964d07bc4d60",
                "TranscriptAnnotator.pm:md5,1afa38e8a2e33fcea46f56661f8013e4",
                "UTRAnnotator.pm:md5,70ec0774faee84ade3a35a1ce07ce701",
                "VARITY.pm:md5,d21a5d6a4022642d13d7c8f8a68fb3b7",
                "Wildtype.pm:md5,685ad5086e3f96ec92127dec78e4aecb",
                "condel_SP.conf:md5,ec4e94fcbb1daff9bbdeaa0005bf7533",
                "polyphen.data:md5,25051ec855d837bb6b175de9b08259a4",
                "sift.data:md5,8d7ce5790192cc979c1f9700be8c9d08",
                "test_condel_SP.data:md5,aced36d2410eeb4485fb499ff9c699f5",
                "cpanfile:md5,e562761d9987b0f3268b385067869a0c",
                "requirements.txt:md5,86ee0804dc1b9e54fc2d376de9a271e0",
                "ubuntu-packages.txt:md5,9b62e56826060a7c24fc483c725365a9",
                "dbNSFP.pm:md5,5cd6245c6b68e215074bed3efa36e859",
                "dbNSFP_replacement_logic:md5,2b372d169e77f1752521e3f57662e42e",
                "dbscSNV.pm:md5,3edb7bf432fe12700ba2fac21b26439f",
                "gnomADc.pm:md5,de737618bce3313b60f2c65208357413",
                "mutfunc.pm:md5,3a85e5e262988d9a11497aafe1418454",
                "neXtProt.pm:md5,4a6d48ed6975d584a8ae8b5a1d966ad2",
                "neXtProt_headers.txt:md5,fd995f62243a5d6beab74d74522fd3b6",
                "pLI.pm:md5,972ade305ffb2cee621c483f51ddd778",
                "pLI_values.txt:md5,8e35e21aa84b5976776deda972883130",
                "plugin_config.txt:md5,197e111960a54e63266c09310c50cb3a",
                "satMutMPRA.pm:md5,6dbac858c60d7fc647e97cae23ad0fba"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.4"
        },
        "timestamp": "2025-07-31T10:28:48.406900323"
    }
}